components/eamxx/src/share/util/eamxx_column_ops.hpp

```diff
@@ -116,6 +116,75 @@ class ColumnOps {
                   "Error! Do not use PackSize>1 on GPU.\n");
   }
 
+  // Apply combine operation y(i) = combine(x(i),y(i),a,b)
+  template<CombineMode CM = CombineMode::Update, typename InputProvider, typename ScalarT, typename MT>
+  KOKKOS_INLINE_FUNCTION
+  static void
+  update (const TeamMember& team,
+          const InputProvider& x,
+          const view_1d<ScalarT,MT>& y,
+          const scalar_type& alpha,
+          const scalar_type& beta)
+  {
+    auto lambda = [&](int k) {
+      combine<CM>(x(k),y(k),alpha,beta);
+    };
+    team_parallel_for(team,y.size(),lambda);
+  }
+
+  // Given input x, compute adjacent increment dx via fwd difference dX(i)=x(i+1)-x(i).
+  // NOTE: we ask for x_scalar_len since x may not be a view (e.g., a lambda). Also,
+  // we ask for the SCALAR length (i.e., without any simd type) so that we can correctly
+  // deduce what the length of the loop is
+  template<CombineMode CM = CombineMode::Replace, typename InputProvider, typename ScalarT, typename MT>
+  KOKKOS_INLINE_FUNCTION
+  static void
+  diff (const TeamMember& team,
+        const int x_scalar_len,
+        const InputProvider& x,
+        const view_1d<ScalarT,MT>& dx,
+        const scalar_type alpha = one(),
+        const scalar_type beta = zero())
+  {
+    const int x_len = ekat::PackInfo<pack_size<ScalarT>()>::num_packs(x_scalar_len);
+    const int dx_len = ekat::PackInfo<pack_size<ScalarT>()>::num_packs(x_scalar_len-1);
+    auto lambda = [&](int k) {
+      auto tmp = ekat::adj_diff<true>(x,k,x_len);
+      combine<CM>(tmp,dx(k),alpha,beta);
+    };
+    team_parallel_for(team,dx_len,lambda);
+  }
+
+  // Compute adjacent averages
```

Comment:
I think we should have the same note about `x_scalar_len` here too in case someone only navigates to this function and might be confused.